src/models/CourseModel.js

```diff
@@ -0,0 +1,103 @@
+const admin = require('firebase-admin');
+
+class CourseModel {
```

Comment:
This is way too many things to start with. We can add additional fields to this model as we need to, but supporting all these at once is too much. Let's start with just the basics of what identifies a course and what it contains.

Things I think should stay:
```
courseName: string
description: string
courseImageUrl: string
owner: userID                    // Or however Firebase represents users
```

Things we should add in future commits that focus on a specific feature:
```
lessons: LessonModel // This doesn't exist yet, but the videos and quizzes should belong to a lesson, and a course should have one or more lessons.
```

Things we should possibly consider adding but need to be fleshed out in their representation:
```
points // Is this just a number for what they get when they finish?
typeOfCourse // How is this different than the name?
```

I think we should shy away from the rest. Some of these I highly doubt we will get to in 8 weeks, like feedback and ratings.

Reply:
I will discuss with team and will changes course collection fields.